### ISSUE TITLE
Feature request: relative links

### ISSUE THREAD

**Issue by priyadarshan** [org2blog/org2blog#249]
It would be quite useful to be able to publish relative links with org2blog, that is, urls without the domain name.

It is possible to export such links using org machinery by evaluating:

```
(defun my/export-rel-url (path desc format)
  (cl-case format
    (html (format "<a href=\"%s\">%s</a>" path (or desc path)))
    (latex (format "\href{%s}{%s}" path (or desc path)))
    (otherwise path)))

(org-add-link-type "rel" 'browse-url 'my/export-rel-url)
```

Then, a snippet like

```
[[rel:/test.txt][a relative link]]
```
will be rendered as
```
<p>
<a href="/test.txt">a relative link</a>
</p>
```

Although org export handles it fine, org2blog will throw error:
```
Wrong type argument: dhar-or-string-p, nil
```

Is there any way to leverage org-mode export, using the code above, so to use relative links like `rel:/test.txt` in org2blog as well?

**Comment by grettke** [comment]
That sounds like a nice feature.

> Is there any way

Yes but right now I don't know how.

It is on the list and I can help immediately by offering feedback to changes that you propose.

**Comment by priyadarshan** [comment]
Sorry, for some reason the patched xml-rpc.el was not loaded.

Now that is is, I am able to publish the test snippet, but org2blog makes it relative to the `/wp-content/uploads/2017/09` directory, while it should be relative to the top domain.

So, the above snippet,
```
[[rel:/test.txt][a relative link]]
```
gets published as
```
<a href="https://www.domain.com/wp-content/uploads/2017/09/test.txt">a relative link</a>
```

**Comment by grettke** [comment]
Org2Blog automatically uploads linked files. It uploads them to the attachment directory. The published link is to that upload directory because it expects that file to be there. 

It looks like you don't want that behavior. I'm not sure right now how to do this.

**Comment by priyadarshan** [comment]
A brief update:

I have found out that if one append a slash to a `rel` link, as setup in my initial ticket, the relative link will work as intended.

Example:

```
[[rel:/sysadmin/][sysadmin]]
```

Will correctly be posted to a wp site as 
```
<a href="/sysadmin/">sysadmin</a>
```

To summarise: I can use relative links for everything but not to pinpoint a single file. In that case, I guess `org2blog` machinery interferes with `org` standard export mechanism.

I am studying more elisp trying to fix that last aspect.

**Comment by grettke** [comment]
Thank you for the detail and effort.

We can make relative link exports to WordPress work however makes sense. I'll leave this ticket open until we do.

**Comment by grettke** [comment]
@priyadarshan The reason is that all files specified by a url are getting uploaded to the post and referenced in the upload directory. That clearly isn't your intent. I think I'm going to open a new task and change the uploading behavior.

**Comment by priyadarshan** [comment]
Nice, that would help me a lot. Right now, if needed I am manually writing relative links as html. It would be great to be able to do express relative links within org2blog directly. Thank you for considering this.

**Comment by grettke** [comment]
@priyadarshan Just committed the change. It wasn't so much adding a new feature as tightening up how the existing feature works. It all revolved around the automatic image upload feature. Now the default for that feature is "turned off". If you were using automatic image uploads then you need to turn it on again like by setting `org2blog/wp-image-upload` to a non-nil value. So once you get the updated code I believe that you will get the behavior you are expecting. If you want to read about that upload behavior it is all [here](https://github.com/org2blog/org2blog#automatic-image-uploading). The following are screenshots of how the code works now:

<img width="205" alt="Screenshot 2020-02-29 22 26 46" src="https://user-images.githubusercontent.com/2521647/75619619-1ff0cd00-5b44-11ea-8355-22a0575473a3.png">
<img width="213" alt="Screenshot 2020-02-29 22 27 34" src="https://user-images.githubusercontent.com/2521647/75619621-20896380-5b44-11ea-83d3-ea14f168a441.png">
<img width="249" alt="Screenshot 2020-02-29 22 27 42" src="https://user-images.githubusercontent.com/2521647/75619622-20896380-5b44-11ea-9438-dc37c96443a2.png">
<img width="389" alt="Screenshot 2020-02-29 22 27 50" src="https://user-images.githubusercontent.com/2521647/75619623-2121fa00-5b44-11ea-9a04-e964a91c06ca.png">
<img width="170" alt="Screenshot 2020-02-29 22 27 57" src="https://user-images.githubusercontent.com/2521647/75619625-2121fa00-5b44-11ea-8a4b-4af1ac673925.png">
<img width="283" alt="Screenshot 2020-02-29 22 28 04" src="https://user-images.githubusercontent.com/2521647/75619627-2121fa00-5b44-11ea-9b91-05c6ff3c5464.png">
<img width="285" alt="Screenshot 2020-02-29 22 28 09" src="https://user-images.githubusercontent.com/2521647/75619628-21ba9080-5b44-11ea-83da-21deea1cf9f0.png">

Please try it out and let me know what you think.

If you are satisfied with the change then please close the issue or I can do it for you.

Let me know how it goes.

**Comment by grettke** [comment]
@priyadarshan I didn't tag it yet so if you are using MELPA unstable you will get the most current version soon otherwise you can download it here. Let me know if I can help in that process.

**Comment by priyadarshan** [comment]
Thank you, this is perfect. I was able to test it via git source code directly. Thank you also for the fantastic job on the documentation. It is inspiring to see such care, quality and precision. Thank you!

**Comment by grettke** [comment]
@priyadarshan You are welcome. Thank you for helping Org2Blog move forward. Happy blogging!

**Comment by grettke** [comment]
@priyadarshan Just so you know I had to make a bug fix just now. I had accidentally disabled posting entirely! It is resolved now in v1.1.8. Sorry for any trouble it caused. Let me know if you run into any issues. Happy Blogging.
